protobuf/cluster/cluster.proto

```diff
@@ -0,0 +1,30 @@
+//
+// Copyright (C) 2021 Grakn Labs
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+syntax = "proto3";
+
+option java_package = "grakn.protocol.cluster";
+option java_outer_classname = "ClusterProto";
+
+import "protobuf/cluster/discovery.proto";
+
+package grakn.protocol;
+
+service Cluster {
+    rpc cluster_discovery (Discovery.Cluster.Req) returns (Discovery.Cluster.Res);
```

Comment:
I'm not a huge fan of these method names. In Core, we use an object and an optional verb: `session_open`, `database_delete`, etc. Do you think we should use `cluster_discover` and `database_replicas_discover` (and rename message `Discovery` to `Discover`)?

Reply:
Actually I'm not sure this is the right grouping either. Doesn't it feel like `Cluster` and `DatabaseReplica` should be top-level messages, and `Discover` should be their child?

Reply:
> I'm not a huge fan of these method names. In Core, we use an object and an optional verb: session_open, database_delete, etc. Do you think we should use cluster_discover and database_replicas_discover (and rename message Discovery to Discover)?

I agree with this. I think we should update the name to be `cluster_discover` and `database_replicas_discover`.

Reply:
> Actually I'm not sure this is the right grouping either. Doesn't it feel like Cluster and DatabaseReplica should be top-level messages, and Discover should be their child?

I kind of like that as well, because then we don't have to introduce too many new top-level terminology. "discovery" is just an operation on existing concepts such as cluster / servers and database.